src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -562,21 +503,26 @@ private static void writeDataToFile(IonType type, IonWriter writer, Random rando
                 break;
             case STRING:
                 List<Integer> pointRange = (List<Integer>) objects[4];
-                writer.writeString(WriteRandomIonValues.constructString(pointRange));
+                writer.writeString(WriteRandomIonValues.constructString(pointRange, 20));
                 break;
             case DECIMAL:
                 List<Integer> expValRange = (List<Integer>) objects[4];
                 List<Integer> coefficientDigitRange = (List<Integer>) objects[5];
                 writer.writeDecimal(WriteRandomIonValues.constructDecimal(expValRange, coefficientDigitRange));
                 break;
             case INT:
-                WriteRandomIonValues.constructInt(writer, random);
+                WriteRandomIonValues.constructInt(random, writer);
                 break;
             case TIMESTAMP:
                 Object timestampTemplate = objects[4];
                 Timestamp timestamp = WriteRandomIonValues.constructTimestamp(timestampTemplate);
                 writer.writeTimestamp(timestamp);
                 break;
+            case STRUCT:
+                IonStruct fields = (IonStruct) objects[4];
+                IonList annotations = (IonList) objects[5];
```

Comment:
I know this wasn't changed in this PR (and I probably should have caught it before), but we shouldn't be using `Object... objects` in method signatures to allow the method to be provided with any arguments. It leads to too many lines of code like this, which are hard to change if the implicit spec for what's allowed in the `objects` array is changed. I recommend creating named parameters for everything in the method signatures, even if some of the parameters are unused by some of the types (this change will affect several of the methods in this class).

Reply:
> I know this wasn't changed in this PR (and I probably should have caught it before), but we shouldn't be using `Object... objects` in method signatures to allow the method to be provided with any arguments. It leads to too many lines of code like this, which are hard to change if the implicit spec for what's allowed in the `objects` array is changed. I recommend creating named parameters for everything in the method signatures, even if some of the parameters are unused by some of the types (this change will affect several of the methods in this class).

Changes have been updated.